src/thunderbots/software/util/canvas_messenger/canvas_messenger.cpp

```diff
@@ -130,6 +148,45 @@ void CanvasMessenger::drawRectangle(Layer layer, Rectangle rectangle, Angle orie
     drawSprite(layer, rectangle_sprite);
 }
 
+void CanvasMessenger::drawGradient(Layer layer, std::function<double(Point)> f,
+                                   const Rectangle& area, double min_val, double max_val,
+                                   Color min_color, Color max_color, int points_per_meter)
+{
+    for (int i = 0; i < area.width() * points_per_meter; i++)
+    {
+        for (int j = 0; j < area.height() * points_per_meter; j++)
+        {
+            Point p = area.swCorner() +
+                      Vector(0.5 / points_per_meter, 0.5 / points_per_meter) +
+                      Vector(i / (double)points_per_meter, j / (double)points_per_meter);
```

Comment:
preferably use `static_cast`

Reply:
Changed here and in a few other spots in this file.